[Lei Jianwen] iP

Jianwen0451 opened the pull request:
Kotori

> “ _Even newborn baby birds
> Will someday soar through the sky
> They'll fly with great, strong wings_
> .” – µ’s ([source](https://genius.com/Genius-english-translations-s-start-dash-english-translation-lyrics))

Kotori Minami is now in your computer to help you keep track of your task list 
It's,

- text-based
- simple to use, can be learned by monkeys 🐵
- completely **FREE**!
- coded in Java

All you need to do to enjoy this bot is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let Kotori manage your tasks for you 😉

Features:

- [x]  Managing tasks
- [ ]  A better GUI (coming soon)
- [ ]  Add a picture/ short video of Kotori (coming soon)

If you are a Java programmer, you can use it through a simple java programme. Here's the main method:
```Java
public class Main {
    public static void main(String[] args) {
        new Kotori().run()
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jianwen0451 committed:
Done Level 0, the bot is renamed to Kotori

Jianwen0451 committed:
Level 0 is done, chatbot is renamed  to Kotori

Jianwen0451 committed:
Level 1 done, The chatbot is now able to echo the input from the command line and exit when seeing bye

Jianwen0451 committed:
Level 2.0 is completed, Now the chatbot is able to keep track of a list of events.

Jianwen0451 committed:
Fix the bugs for adding into list.

Jianwen0451 committed:
Add the class Task.java and implement the features to mark and unmark. Level 3 done

Jianwen0451 committed:
Level 4 complete, now task has 3 subtype as event, deadline and todo

Jianwen0451 committed:
Add a test case to test the bot Using I/O redirection. A-TestUiTesting is finished

Jianwen0451 committed:
Change the bug of One extra white space

Jianwen0451 committed:
Level 5 complete, the bot is now able to deal with some invalid inputs like missing element and unknown command. Add a few test cases to test the error handling.

Jianwen0451 committed:
Fix some bugs on the test cases

Jianwen0451 committed:
Level 5 complete. Add deleting operation. Add a error handling cases about deleting. Add test case for deleting operation.

Jianwen0451 committed:
A-enum done. Enum is used in IncorrectStateException.java

Jianwen0451 committed:
Fix minor bugs in formatting

Jianwen0451 committed:
Replace String[] with String...

Jianwen0451 committed:
Complete Level 7

This is to follow the instruction of Individual project

It will create a directory and a txt file if needed and store the list inside the file after each operation.

It is to save the content of the list for future use.

Jianwen0451 committed:
Level 7 is now finally complete. The bot is able to save and read successfully

This is to follow the instruction of Individual project

It will create a directory and a txt file if needed and store the list inside the file after each operation.

It is to save the content of the list and load it when running again

Jianwen0451 committed:
Remove the files that should be ignored

Jianwen0451 committed:
Ignore Kotori.txt and text-ui-test/data/Kotori.txt

They are ignored because they are local data

Add Kotori.txt and text-ui-test/data/Kotori.txt into .gitignore

Jianwen0451 committed:
Ignore Kotori.txt and text-ui-test/data/Kotori.txt

Jianwen0451 committed:
Complete Level 8 Add feature of Date Management

To manage the by, from, and to as real date rather than String

LocalDate class is introduced as instructed.
A new command find is created that all the deadline and event that need to do at this time

Jianwen0451 committed:
Merge tag 'Level-7'

* tag 'Level-7':
  Ignore Kotori.txt and text-ui-test/data/Kotori.txt
  Remove the files that should be ignored
  Level 7 is now finally complete. The bot is able to save and read successfully
  Complete Level 7

Jianwen0451 committed:
Merge branch 'Level-8'

* Level-8:
  Complete Level 8 Add feature of Date Management

# Conflicts:
#	src/main/java/Kotori.java
#	src/main/java/Task.java
#Conflicts resloved

Jianwen0451 committed:
Complete MoreOOP and Packaging

Follow the instructions and do the stretched goal

Create a few command as OOP, create the Ui, Storage and Tasklist class to follow OOP

Jianwen0451 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Jianwen0451 committed:
Complete A-Graddle, manage to set up gradle for the project

Merge the branch Bradle help with the master branch and set up and mainclass

Jianwen0451 committed:
Fix some bugs about level-7 and level-8 about implement OOP

Save the file to hard disk whenever there is a modification to the list

Jianwen0451 committed:
Fix a bug about level-8 packages

Put all the sub-packages and Kotori.java in to a package called Kotori

Jianwen0451 committed:
Complete A-JUnit with strech goals and fix some minor issue

Include the Junit
Create test cases for all non-trivial methods in Command, Parser, and Storage
Change the name of all packages to all small cases

Jianwen0451 committed:
Complete A-jar

Create the jar file based on the tution

Jianwen0451 committed:
Complete A-JavaDoc Create JavaDoc for all non-trival class and methods. both public and private

Write JavaDoc based on the requirement

Jianwen0451 committed:
Change the style to follow the coding standard

Jianwen0451 committed:
Complete Level-9 Add the find command as instructed. The original find command is renamed to search command

Add hasDescription(String input) method in Task
Add findAll(String input) method in TaskList
Add a new FindCommand
Update parser

Jianwen0451 committed:
Merge tag 'A-CodingStandard' Conflict resolved

* tag 'A-CodingStandard':
  Change the style to follow the coding standard

# Conflicts:
#	src/main/java/kotori/command/AddCommand.java
#	src/main/java/kotori/command/Command.java
#	src/main/java/kotori/command/ExitCommand.java
#	src/main/java/kotori/command/searchCommand.java
#	src/main/java/kotori/storage/Storage.java
#	src/main/java/kotori/taskList/TaskList.java

Jianwen0451 committed:
Merge branch 'Level-9' Resolved conficts

* Level-9:
  Complete Level-9 Add the find command as instructed. The original find command is renamed to search command

# Conflicts:
#	src/main/java/kotori/Ui/Ui.java
#	src/main/java/kotori/command/searchCommand.java
#	src/main/java/kotori/parser/Parser.java
#	src/test/java/KotoriTest.java
#	testData/Add.txt

Jianwen0451 committed:
Change the name of package Ui to ui

Jianwen0451 committed:
No change done, this commit is trying to deal with some technical difficulty

Jianwen0451 committed:
Solve the the difficulties and get the SearchCommand class back

Jianwen0451 committed:
Fix some minor issue on coding-standard

Jianwen0451 committed:
Add use of trim() to parser for adapting to more command line input

Jianwen0451 committed:
Add basic of GUI. The bot is able to do echo the input of user now

This is done by following through the GUI guide and implement a fxml file

Jianwen0451 committed:
Complete Level-10, the UI of bot is complete

Move the logic of print into UI and fxml files

Jianwen0451 committed:
Use VarArgs in the bot to achieve the optional goal

VarArgs is used in printMessages and printListWithMessages methods

SherChew27 commented on src/main/java/kotori/command/Command.java:
```diff
@@ -0,0 +1,11 @@
+package kotori.command;
```
well done on abstracting the command class from the parser class!

SherChew27 commented on src/main/java/kotori/taskList/MissingInformationException.java:
```diff
@@ -0,0 +1,24 @@
+     * @return the error message.
+     * */
+    public String getMessage() {
+        return String.format("Sorry~ Please enter the %s for %s", missingInfo, type);
```
good exception handling, anticipating the various potential exceptions that can be thrown

SherChew27 reviewed (commented):
Very well done!!

Jianwen0451 committed:
Implement checkstyle and fix some incorrect style problem

Arren11111 commented on src/main/java/kotori/Kotori.java:
```diff
@@ -0,0 +1,88 @@
+/**
+ * This class represents the bot.
+ * */
+
```
Javadoc comments should not leave a line before the class or method it is describing

Arren11111 commented on src/main/java/kotori/Kotori.java:
```diff
@@ -0,0 +1,88 @@
+
+
+/**
+ * This class represents the bot.
```
Should start the comment with "Represents..."

Arren11111 commented on src/main/java/kotori/Kotori.java:
```diff
@@ -0,0 +1,88 @@
+    private Parser parser;
+
+    /**
+     * Produce a bot.
```
Verbs should be of the form XXXs, in this case, it should be "Produces" instead of "Produce"

Arren11111 commented on src/main/java/kotori/Kotori.java:
```diff
@@ -0,0 +1,88 @@
+    /**
+     * Let the bot start running.
+     * */
+
```
Blank line should be removed

Arren11111 commented on src/main/java/kotori/Kotori.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    /**
+     * Return a string for Ui.
```
```suggestion
     * Returns a string for Ui.
```

Arren11111 commented on src/main/java/kotori/Kotori.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    /**
+     * Return a string of reading status for Ui.
```
```suggestion
     * Returns a string of reading status for Ui.
```

Arren11111 commented on src/main/java/kotori/command/AddCommand.java:
```diff
@@ -0,0 +1,55 @@
+ * This class represents a command of trying to create
+ * a task and add it into the task list.
+ * */
+
```
```suggestion
```

Arren11111 commented on src/main/java/kotori/command/AddCommand.java:
```diff
@@ -0,0 +1,55 @@
+    private String input;
+
+    /**
+     * Create an add-command
```
```suggestion
     * Creates an add-command
```

Arren11111 commented on src/main/java/kotori/command/AddCommand.java:
```diff
@@ -0,0 +1,55 @@
+     * Activates the command the try to create
+     * and add a new task to the task list.
+     * */
+
```
```suggestion
```

Arren11111 commented on src/main/java/kotori/command/Command.java:
```diff
@@ -0,0 +1,11 @@
+ * This is an abstract class representing
+ * a command.
+ * */
+
```
```suggestion
```

Arren11111 commented on src/main/java/kotori/command/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+ * This class represents a command of trying to
+ * delete the task at a certain position of the task list.
+ * */
+
```
```suggestion
```

Arren11111 commented on src/main/java/kotori/command/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+    private int index;
+
+    /**
+     * Create a delete command
```
```suggestion
     * Creates a delete command
```

Arren11111 commented on src/main/java/kotori/command/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * Activates the command and try to delete the task.
+     * */
+
```
```suggestion
```

Arren11111 commented on src/main/java/kotori/command/ExitCommand.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * This class represents a command of exiting the chatbot.
+ * */
+
```
```suggestion
```

Arren11111 reviewed (commented):
Needs to format Javadoc comments appropriately according to course conventions

Jianwen0451 committed:
Add assert function into the Storage object to check that
there is a actual file in the Storage

This is to follow the instruction of the course and make the code safer

Add an assertion line when creating new instance of Storage item

The code should not continue further if there is no memory file and fail to create.
The assertion is to check that the code is continue with a memory file.

Jianwen0451 committed:
Do the changes on some magic literals to improve code quality

This is to follow the instruction and make my code more readable

Separate a lot of literals from methods and give them a proper name to
increase readability.

This is to ensure that myself and other editors are able to read and understand the
functionality of the code.

Jianwen0451 committed:
Add a new keyword sort to sort the task list, with the task having close deadline/ to date in front

This is done as a requirement of course as the BCD-increment.

A new command SortCommand is added. This command will sort the list and store the result new list
into the storage.

Jianwen0451 committed:
Merge pull request #1 from Jianwen0451/A-Assertions

Add assert function into the Storage object to check that

Jianwen0451 committed:
Merge branch 'master' of https://github.com/Jianwen0451/ip into A-Stream

* 'master' of https://github.com/Jianwen0451/ip:
  Add assert function into the Storage object to check that there is a actual file in the Storage

Jianwen0451 committed:
Merge branch 'master' into A-CodeQuality

* master:
  Add assert function into the Storage object to check that there is a actual file in the Storage

# Conflicts:
#	src/main/java/kotori/Kotori.java

Jianwen0451 committed:
Merge pull request #2 from Jianwen0451/A-CodeQuality

A code quality

Jianwen0451 committed:
Merge branch 'master' into A-Stream

* master:
  Do the changes on some magic literals to improve code quality

Jianwen0451 committed:
Merge pull request #3 from Jianwen0451/A-Stream

A stream

Jianwen0451 committed:
Update the Junit test and implement SLAP on Task class

This is to follow the requirement of IP

Update the Junit test to let it fit in current version of Kotori
Update the Task class so that Task.of method adhere SLAP better.

Jianwen0451 committed:
Remove file Temp and update the GUI

The file temp is unnecessary and the GUI is not able to change size.

Removal of garbage files will prevent unnecessary overheads.
GUI is updated to change its sizes accordingly.

Let's try
* The new GUI!

Jianwen0451 committed:
Add the bot name Kotori to the title bar of GUI

GUI does not have the bot name on its title bar.

This is to make the name of the bot more obvious to the user.

Let's see the updated GUI!

Jianwen0451 committed:
Separate the ToDo class Deadline class and Event class from Task class

The 3 subclass are originally composited with Task class.

They are separated because they have different behaviours.

Let's create cleaner code!

Jianwen0451 committed:
Add error handling to mark and unmark command with non-number input
like "mark ???"

Mark and unmark are not able to handle such input.

The bot now can deal with more invalid input.

Let's make the bot smarter!

Jianwen0451 committed:
Create website for chatbot Kotori

There is no website for my bot.

The bot is now illustrated by the Readme.md and UI.png.

Jianwen0451 committed:
Remove a few magic number

There are some magic numbers in MarkCommand and UnmarkCommand.

Make the program more readable.

Jianwen0451 committed:
Remove the jar file and rename package taskList to tasklist

There is a ip.jar in repo and package name should be in all lower
cases.

This is to follow the specification of the course.

Jianwen0451 committed:
Change the webpage file to make the User Guide more detailed

The user guide before is lack of example of how the commands
are used and example output.

Jianwen0451 committed:
Change wildcard import to full import

There is wildcard import in parser and Junit tests.

This is to ensure the readability of the code and
follow the regulation.

Jianwen0451 committed:
Change name of UI.png to Ui.png

The name of the Png file is wrong.

Jianwen0451 committed:
Add example output for sort method

There is no example output for sort method in Readme.

This is make the user guide more useful.

Jianwen0451 committed:
Add error handling for invalid number of input
and fix the error handling for missing description

Such errors are not handled properly before.

Jianwen0451 committed:
Fix some error in the website file

There are typos and wrong description in the website

Jianwen0451 committed:
Fix some error in the website file

There are typos and wrong description in the website.

Jianwen0451 committed:
Fix some error and add more description in the website file

There are typos and wrong description in the website.

Some instructions are not well explained.

Jianwen0451 committed:
Fix a naming of boolean and separate out the condition for
add command for better code quality

There is a naming and a long conditional statement
that do not follow the code quality.

Jianwen0451 committed:
Change the output format of time expression in tasks

The current expression of time is in the same format.

It is changed to meet the requirement of the ip.

Jianwen0451 committed:
Update UI tests and website to be consistent with the new format
of time

The current time format is not consistent with tests and website

Jianwen0451 committed:
Change the Ui picture so that it is consistent with the
new time format

The Ui picture is not consistent with the new time format

Jianwen0451 committed:
Update the example output in the user guide

The example output in user guides is a little bit different from the
actual output

Jianwen0451 committed:
Update User Guide format

There are some format errors in the user guide

Jianwen0451 committed:
Update the user guide

Some of the example output is not the same as actual output
Some description of command is not clear enough

Jianwen0451 committed:
Add more clarification in user guide

There is not enough explanation about search command in
user guide
